server/push_notifications.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"fmt"
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+func (p *Plugin) NotificationWillBePushed(notification *model.PushNotification, userID string) (*model.PushNotification, string) {
```

Comment:
I now realize that we should have probably named this `PushNotificationWillBeSent`.

Reply:
In retrospect, yah. But it still matches with the other noun-will-be-verbed methods, and it's kind of interesting.